scalyr_agent/monitor_utils/k8s.py

```diff
@@ -1310,30 +1581,56 @@ def query_api( self, path, pretty=0, return_temp_errors=False ):
             pretty = '?%s' % pretty
 
         url = self._http_host + path + pretty
+
         response = None
+
+        # Optionally save api json to disk
+        log_responses = self.log_api_responses_to_disk and self.agent_log_path
+        logged_response_file = None
+        if log_responses:
+            logged_response_file = self.__open_api_response_log(path, rate_limited)
         try:
-            response = self._session.get( url, verify=self._verify_connection(), timeout=self.query_timeout )
-            response.encoding = "utf-8"
-        except Exception, e:
-            if return_temp_errors:
-                raise K8sApiTemporaryError('Temporary error seen while accessing api: %s' % str(e))
-            else:
-                raise
+            # Optionally prepend stack trace into logged response file
+            if logged_response_file:
+                traceback.print_stack(file=logged_response_file)
+                logged_response_file.write('\n\n')
 
-        if response.status_code != 200:
-            if response.status_code == 401 or response.status_code == 403:
-                raise K8sApiAuthorizationException( path, status_code=response.status_code )
-            elif response.status_code == 404:
-                raise K8sApiNotFoundException( path, status_code=response.status_code )
-
-            global_log.log(scalyr_logging.DEBUG_LEVEL_3, "Invalid response from K8S API.\n\turl: %s\n\tstatus: %d\n\tresponse length: %d"
-                % ( url, response.status_code, len(response.text)), limit_once_per_x_secs=300, limit_key='k8s_api_query' )
-            if return_temp_errors:
-                raise K8sApiTemporaryError( "Invalid response from Kubernetes API when querying '%s': %s" %( path, str( response ) ), status_code=response.status_code )
-            else:
-                raise K8sApiException( "Invalid response from Kubernetes API when querying '%s': %s" %( path, str( response ) ), status_code=response.status_code )
+            # echee: TODO: remove this before merging
```

Comment:
There are some outstanding TODOs that must be resolved before merging back to master. This is one of them.